go-controller/pkg/util/pod_annotation.go

```diff
@@ -555,6 +575,11 @@ func AddRoutesGatewayIP(
 					podAnnotation.Gateways = append(podAnnotation.Gateways, gatewayIPnet.IP)
 				}
 			}
+			joinAddrs, err := ParseNodeGatewayRouterJoinAddrs(node, netinfo.GetNetworkName())
+			if err != nil {
+				return err
+			}
+			podAnnotation.IPv6LLAGateway = HWAddrToIPv6LLA(IPAddrToHWAddr(joinAddrs[0].IP))
```

Comment:
dont you need to add check like this ?
https://github.com/ovn-kubernetes/ovn-kubernetes/pull/4852/files#diff-01a2ef868473804cd3d9b894b093e5be0735cfb21bf27f375ece3a33cc07edc9R859

(so ipv4 single stack wont have the new annotation even that it is derived from ipv4), else how will CNI know it shouldnt create the tables ?
(unless i am missing something, i am just learning the ropes here)

Reply:
Right, we should not added for single stack ipv4.